src/TextField/TextField.jsx

```diff
@@ -325,7 +325,7 @@ const TextField = React.createClass({
     if (hasValueLinkProp) {
       newState.hasValue = isValid(nextProps.valueLink.value);
     } else if (hasValueProp) {
-      newState.hasValue = isValid(nextProps.value);
+      newState.hasValue = isValid(nextProps.value) || isValid(nextProps.defaultValue);
```

Comment:
Hmm you're right.  The problem is that `componentWillReceiveProps` is being called right after the initial `state` is set for some reason.  This seemed to fix the bug but it isn't correct.

I'll try to come up with a solution.